### PR TITLE
sequential multiple add, hybrid warning

### PR DESCRIPTION
**Pull request by szha**

### CODE REVIEW

File: python/mxnet/gluon/nn/basic_layers.py [hunk 1]
```diff
@@ -94,8 +123,18 @@ def __repr__(self):
...
                         modstr=modstr)
 
-    def __getitem__(self, i):
-        return self._children[i]
+    def __getitem__(self, key):
+        if isinstance(key, integer_types):
+            return self._children[key]
+        elif isinstance(key, py_slice):
+            out = HybridSequential(self._prefix, self._params)
+            for c in self._children[key]:
```
Comment (piiswrong) [inline python/mxnet/gluon/nn/basic_layers.py]: registering one layer as children of multiple layers is a bad idea.

Comment (szha) [reply python/mxnet/gluon/nn/basic_layers.py]: What's the alternative?

Comment (szha) [reply python/mxnet/gluon/nn/basic_layers.py]: Please elaborate

Comment (piiswrong) [reply python/mxnet/gluon/nn/basic_layers.py]: return a list or remove this feature

Comment (mli) [reply python/mxnet/gluon/nn/basic_layers.py]: why it is problematic

Comment (szha) [reply python/mxnet/gluon/nn/basic_layers.py]: With this feature, running a subset of forward will become very easy. Users can simply do
```python
out = model.features[:5](input)
```
I think this simplicity can be useful. If there are problems, let's try to address it together here.

**Comment by szha** [comment]
Any feed back other than dropping the feature?

**Pull request merged**